### PR TITLE
Refactor to use Flask-Rest-x and generate Swagger Docs

### PR DESCRIPTION
**Pull request by mingyi850**
- Augmented flask with flask-restx framework
- All api routes now prefaced with /api
- Added validations for all routes by type
- Added swagger documentation as part of flask-restx
- Remove old error handlers

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -46,123 +38,318 @@ def index():
...
+    @api.response(404, "Promotion not found")
+    def delete(self, promotion_id):
+        """
+        Deletes a Promotion with promotion_id
+        """
+        app.logger.info(f"Got request to delete Promotion with id: {promotion_id}")
+        promotion = Promotion.find(promotion_id)
+        if promotion:
+            promotion.delete()
+
```
Comment (ChungLinWu0829) [inline service/routes.py:255]: Do we also return HTTP_204_NO_CONTENT when the ID does not exist? I reviewed the test cases, and it seems there is some divergence.
image

Comment (mingyi850) [reply service/routes.py:255]: It should be a 204
image

Let me change the spec string. Good catch!

Comment (ChungLinWu0829) [reply service/routes.py:255]: Thanks for clarifying!

**Review by ChungLinWu0829 — APPROVED** [review]
LGTM

File: service/routes.py (hunk 1 above)
Comment (mingyi850) [reply service/routes.py:254]: Got it thanks!

**Pull request merged**